Added write queue depth value

Mavrrick opened the pull request:
This is a simple update to make the write queue customizable

1. Add a setting value that will allow for setting a write depth for the queue
2. Add a check for a null value for the new setting value to ensure backwards compatibility
3. Add the new setting value into the  comparison to initiate a forced write when the queue is full

One thing of note is the default value for the setting is 50 in this update. Further investigation may be needed to tweak this value.

Mavrrick committed:
Added write queue depth value

dennypage commented:
Can we call the preference a "limit" ?

The code can be simplified like this

```
    if (queueSize > settings.prefWriteQueueLimit ?: 100) {
        logger("Queue size is too big, triggering write now", "info")
        writeQueuedDataToInfluxDb()
    }

```

Mavrrick commented:
I have updated my base code i am testing with to use Limit

Perhaps i did something wrong , but when i added that code alone in place  for my changes in the queueToInfluxDb method it wasn't working and writing every record out as if it was set to zero. I added back in my code that validates if the value is null and it is working again. Should that still be needed with your change above? I suspect it would work if i updated the limited value, but this was a test for the upgrade for users when the get it through HPM.

dennypage commented:
I'll have to test it, maybe the operation priority isn't what I think it is and parens are necessary (like below).

The basis is a ternary operator and is pretty common. I use it in some of my other code. These two blocks should be the same:

```
    def limit
    if (settings.prefWriteQueueLimit != null && settings.prefWriteQueueLimit) {
        limit = settings.prefWriteQueueLimit
    } else {
        limit = 100
    }
    if (queueSize > limit) {
        logger("Queue size is too big, triggering write now", "info")
        writeQueuedDataToInfluxDb()
    }
```

```
    if (queueSize > (settings.prefWriteQueueLimit ?: 100)) {
        logger("Queue size is too big, triggering write now", "info")
        writeQueuedDataToInfluxDb()
    }
```

Good explanation [here](https://www.danvega.dev/blog/2013/08/22/groovy-ternary-operator/)

Mavrrick commented:
Yea.. The parentheses are were the issue was. It works as expected. I will update my master now with the changes you have suggested.

Mavrrick committed:
Pull Req asked changes applied

dennypage commented:
Sorry about that.

Mavrrick committed:
Pressure Sensor input bug

Update Pressure sensor input to look at capability.pressureMeasurement instead of capability.sensor.
